CMakeLists.txt

```diff
@@ -966,6 +967,13 @@ if(HAVE_OPENVX)
   endif()
 endif()
 
+if(WITH_KLEIDICV)
+  ocv_debug_message(STATUS "Enable KleidiCV acceleration")
+  if(NOT ";${OpenCV_HAL};" MATCHES ";kleidicv;")
+    set(OpenCV_HAL "kleidicv;${OpenCV_HAL}")
```

Comment:
Please put `if(WITH_KLEIDICV)` back above `if(WITH_CAROTENE)`, as it was before.

`WITH_KLEIDICV` should default to `OFF` before this patch lands, but for the purposes of CI it's interesting to see what builds with it `ON`.

Since `WITH_KLEIDICV` will default to `OFF` and `WITH_CAROTENE` will default to `ON`, it's better that `WITH_KLEIDICV` takes precedence when you do explicitly set it to `ON`.